### DISCUSSION TITLE
How to overwrite parameters?

### DISCUSSION THREAD

**Discussion by raphael-boegel-pcg** [spulec/moto#7696] [Q&A]
Is there anyway to overwrite specific parameters?
For example I want to to overwrite **MaxKeys** of list_objects_v2 (s3).

**Comment by bblommers** [comment]
Hi @raphael-boegel-pcg, are you referring to the default values (e.g. MaxKeys=1000 if not supplied)? Those cannot be changed.

**Reply by raphael-boegel-pcg** [reply]
Hi @bblommers, it is more about the limits than the default values. If you want to check if your app can handle this particular limit, you would have to create more than 1000 fake objects. If you want to check for the handling of the max object size (not multipart) you would have to "upload" more than 100 MB. While possible it would be nice to downscale those limits for testing.

**Reply by bblommers** [reply]
Gocha, that makes sense. I've created an new feature request for this: #7707

**Comment by bblommers** [comment]
This can now be configured as of moto >= 5.0.9.dev2 with an environment variable:
```
MOTO_S3_DEFAULT_MAX_KEYS=1
```

This is documented here: http://docs.getmoto.org/en/latest/docs/services/s3.html
